### PR TITLE
Install latest git to buildbox

### PR DESCRIPTION
**Pull request by gzigzigzeo**
1. GitHub Actions requires git 2.18+ to make `actions/checkout@v3` work. Since, our tests run in Docker containers, we need to have the latest git installed inside them.
2. `test-helm` step has the statement, which sets `HELM_PLUGINS` env var in case the Makefile runs on CI. This PR modifies the condition to make it work with GHA as well.

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

File: build.assets/Dockerfile-centos7 [hunk 1]
```diff
@@ -15,6 +15,9 @@ RUN yum groupinstall -y 'Development Tools' && \
         git \
         libudev-devel \
         zlib-devel && \
+    yum -y remove git \
```
Comment (r0mant) [inline build.assets/Dockerfile-centos7:18]: Should we just not install `git` above on line 15?

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -271,5 +271,10 @@ RUN cd /usr/local/lib && \
     ldconfig
 COPY pkgconfig/buildbox/ /
 
+# Install latest git. git 2.18+ is required for GitHub Actions.
+RUN apt-get -y update && apt-get -y install software-properties-common && \
```
Comment (r0mant) [inline build.assets/Dockerfile:275]: `git` is installed above on line 103 from the repositories - should it be removed from there?

**Review by r0mant — COMMENTED** [review]
Got couple of comments on the second pass.

**Review by russjones — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Comment by r0mant** [comment]
This will be merged in a buddy PR: https://github.com/gravitational/teleport/pull/18498.

**Pull request closed without merging**